### ISSUE TITLE
Pull fails if repository has no latest tag

### ISSUE THREAD

**Issue by jwalant** [docker/kitematic#572]
Sorry for editing @jwalant's original post: 
1. Users get an error if their repository doesn't have the `latest` tag. 

Temp solution: 
- Add a latest tag. 

What we should do: 
- Enable kitematic to check, and pull other versions.

**Comment by jeffdm** [comment]
Taking a look. Thanks for the issue1

**Comment by jeffdm** [comment]
@jwalant are you able to pull these on the command line? I get:

```
$ docker pull netflixoss/hello-netflix-oss
Pulling repository netflixoss/hello-netflix-oss
FATA[0004] Tag latest not found in repository netflixoss/hello-netflix-oss
```

**Comment by jwalant** [comment]
Same here I am not able to pull them.

**Comment by jeffdm** [comment]
Seems like there's no `latest` tag on these repos.

Kitematic should fetch the tags before pulling and default to the first one if `latest` doesn't exist.

Will also investigate for the Docker CLI + Index

**Comment by mchiang0610** [comment]
Yes, there is no latest tag. 

![image](https://cloud.githubusercontent.com/assets/3325447/7965702/03deb35a-09d4-11e5-8bb8-611b164c1a5c.png)

Changing this issue to be a feature request, where Kitematic should prompt the user regarding which tag to use if no latest is found.

**Comment by mchiang0610** [comment]
Kitematic should prompt the user regarding which tag to use if no latest is found.

**Comment by psychemedia** [comment]
+1 on this needing some sort of resolution. I guess the UI requirements are:

- default is `latest`
- allow user to select an arbitrary tag instead of latest
- offer some sort of alternative in the case of going with default an `latest` not being available?

Would one option be to have a drop down list of tags displayed when an image is selected, defaulting to `latest` (if it exists) or most recent tag if it doesn't?

**Comment by FrenchBen** [comment]
Ideally the Registry API would include the most-recent tag as part of their response. This would allow us to set the 'default' as the tag provided by the registry hub. 

The docker command line could also benefit from this, since the following doesn't work:
```bash
$ docker pull netflixoss/hello-netflix-oss
Pulling repository netflixoss/hello-netflix-oss
FATA[0003] Tag latest not found in repository netflixoss/hello-netflix-oss
```

**Comment by jischebeck** [comment]
There are some containers which will never have a latest tag. In addition to the container with the real tag (loaded through the command line), kitematic also shows a cloud symbol for the container with the latest tag and the following error:
    no such container
    404
    No such image: andypetrella/spark-notebook:latest (tag: latest)

**Comment by thaJeztah** [comment]
> Kitematic should fetch the tags before pulling and default to the first one if latest doesn't exist.

and;

> Ideally the Registry API would include the most-recent tag as part of their response. This would allow us to set the 'default' as the tag provided by the registry hub.

The most recent tag may not be the most recent release of an image (e.g. `:v1-legacy` being pushed/built after `:v2-brand-new` was)

TL;DR; I'd be very careful here; I think that asking the *user* to make the decision would really be the best option here, not "automagically" trying to decide

**Comment by abhinavkorpal** [comment]
And the issues still exit !!! Tag issue which I'm already facing.

**Comment by FrenchBen** [comment]
@abhinavkorpal Same thing happens at the CLI level. Feel free to open an issue with the image owner and letting them know how frustrating it is that they don't have a "latest" tag.

**Comment by abhinavkorpal** [comment]
@FrenchBen Sure Will do
